Adding the ability to control looping

davidgeere opened the pull request:
I like the simplicity of this library. I do though think it could use the ability for a developer to set if it never loops, always loops or loops n times. So I added some code that if you add the attribute as such:

- It will loop for ever (it's also the default)
   data-loop="true"

- It will not loop at all, it will just type out once.
  data-loop="false"

- It will loop 4 times
  data-loop="4"

This unfortunately only works with non-delimited words, mainly because multiple delimitations was just not a use case for me. I will submit another change once I get to that.

Commit:
Adding the ability to control looping

I like the simplicity of this library. I do though think it could use the ability for a developer to set if it never loops, always loops or loops n times. So I added some code that if you add the attribute as such:

It will loop for ever (it's also the default)
```
data-loop="true"
```

It will not loop at all, it will just type out once.
```
data-loop="false"
```

It will loop 4 times
```
data-loop="4"
```

This unfortunately only works with non-delimited words, mainly because multiple delimitations was just not a use case for me. I will submit another change once I get to that.

Commit:
Update typer.js

I forgot that false would come through as a string not a boolean type.

straversi commented on typer.js:
```diff
@@ -5,9 +5,10 @@ var Typer = function(element) {
   var words = element.dataset.words || "override these,sample typing";
   this.words = words.split(delim).filter(function(v){return v;}); // non empty words
   this.delay = element.dataset.delay || 200;
+  this.loop = element.dataset.loop || true;
```
You updated to string "false" on line 65, and I think this one should be updated as well to ... || "true".

straversi commented:
Thank you for making this improvement, I love it! I had just one comment before I want to pull the change.

Commit:
Update typer.js

Changed loop property to consistently work with string instead of boolean / string hybrid which could have caused unexpected behavior.

straversi commented:
Thanks for updating-- `since true == 1`, `this.loop <= p.looped` was activating if dataset-loop was not set. Since string < number is not true, the new version is fine.

davidgeere commented:
Awesome, thanks Steven! Thanks for creating this library in the first place. :)
